File: robocode/robocode/battle/events/TurnEndedEvent.java
package robocode.battle.events;


import robocode.battle.snapshot.BattleSnapshot;


public class TurnEndedEvent extends BattleEvent {

	private final BattleSnapshot battleSnapshot;
	
	public TurnEndedEvent(BattleSnapshot battleSnapshot) {
		super();

		this.battleSnapshot = battleSnapshot;
	}

	public BattleSnapshot getBattleSnapshot() {
		return battleSnapshot;
	}
}

File: plugins/dotnet/robocode.dotnet.napi/src/control/events/TurnEndedEvent.cs
using System.Runtime.CompilerServices;
using IKVM.Attributes;
using robocode.control.snapshot;

namespace robocode.control.events
{
    public class TurnEndedEvent : BattleEvent
    {
        [Modifiers(Modifiers.Private | Modifiers.Final)] private ITurnSnapshot turnSnapshot;

        [MethodImpl(MethodImplOptions.NoInlining), LineNumberTable(new byte[] {0x9f, 0xb7, 0x68, 0x67})]
        public TurnEndedEvent(ITurnSnapshot turnSnapshot)
        {
            this.turnSnapshot = turnSnapshot;
        }

        public virtual ITurnSnapshot getTurnSnapshot()
        {
            return turnSnapshot;
        }
    }
}